editoast/src/views/paced_train.rs

```diff
@@ -202,19 +211,62 @@ struct SimulationBatchForm {
 )]
 async fn simulation_summary(
     State(AppState {
-        db_pool: _db_pool,
-        valkey: _valkey_client,
-        core_client: _core,
+        db_pool,
+        valkey: valkey_client,
+        core_client: core,
         ..
     }): State<AppState>,
-    Extension(_auth): AuthenticationExt,
+    Extension(auth): AuthenticationExt,
     Json(SimulationBatchForm {
-        infra_id: _infra_id,
-        electrical_profile_set_id: _electrical_profile_set_id,
-        ids: _paced_train_ids,
+        infra_id,
+        electrical_profile_set_id,
+        ids: paced_train_ids,
     }): Json<SimulationBatchForm>,
 ) -> Result<Json<HashMap<i64, SimulationSummaryResult>>> {
-    todo!();
+    let authorized = auth
+        .check_roles([BuiltinRole::InfraRead, BuiltinRole::TimetableRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+
+    let infra = Infra::retrieve_or_fail(conn, infra_id, || PacedTrainError::InfraNotFound {
+        infra_id,
+    })
+    .await?;
+
+    let paced_trains: Vec<PacedTrain> =
+        PacedTrain::retrieve_batch_or_fail(conn, paced_train_ids, |missing| {
+            PacedTrainError::BatchPacedTrainNotFound {
+                count: missing.len(),
+            }
+        })
+        .await?;
+    let paced_trains_to_ts: Vec<TrainSchedule> =
+        paced_trains.clone().into_iter().map_into().collect();
```

Comment:
This means we're always simulating at the time of the first occurrence right?

Reply:
Yes, if the front needs the following occurrences, it must simulate the following trains by adding the step to the train start_time.

Reply:
Maybe a comment would be nice to actually state the non-obvious here 🤷

Reply:
Ok, how about not using `From` and `into` for this case? Since this is a rather strong semantic change, maybe a function `into_first_occurrence(self) -> TrainSchedule` would be more explicit?